### PR TITLE
Added support for Linear Gradient Prop "gradientUnit" & Changes to compatible with codegen tool

### PR DESCRIPTION
**Pull request by shefayaparwin**
# Summary

- Handled gradientUnit **objectBoundingBox**  && **userSpaceOnUse**
- **Added changes to enable react native svg compatible with RSk code gen tool**
- Removed Manual Integration Guide , as it is no more needed with codegen support
- Added rnsvg modules "RNSVGRenderableModule" && "RNSVGViewModule". None of the functionality were implemented as it is not used in framework. Just Added to fulfill the basic requirement.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| ubuntu     |    ✅ |
| iOS     |    ❌     |
| Android |    ❌     |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added documentation in `README.md`
- [ ] I updated the typed files (typescript)
- [ ] I added a test for the API in the `__tests__` folder

### CODE REVIEW

File: skia/RNSVGRenderableModule.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        return true;
+      }),
+  };
+}
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+RNS_EXPORT_MODULE(RNSVGRenderableModule)
```
Comment (munezbn) [inline skia/RNSVGRenderableModule.cpp]: RNS_EXPORT_MODULE_WITHOUT_SUFFIX

File: skia/Rsk.project [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+        "RNSVGPath",
+        "RNSVGRect",
+        "RNSVGSvgView",
+        "RNSVGText",
+        "RNSVGTSpan",
+        "RNSVGUse"
+      ],
+      "TurboModuleRegistryGetEnforcingSpec" : [
+        "RNSVGRenderable",
+        "RNSVGSvgView"
```
Comment (munezbn) [inline skia/Rsk.project]: "RNSVGRenderableModule",
  "RNSVGSvgViewModule"

File: skia/RNSVGSvgViewModule.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        RNS_UNUSED(this);
+        return;
+      }),
+  };
+}
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+RNS_EXPORT_MODULE(RNSVGSvgViewModule)
```
Comment (munezbn) [inline skia/RNSVGSvgViewModule.cpp]: RNS_EXPORT_MODULE_WITHOUT_SUFFIX

File: skia/RSkComponentRNSVGLinearGradient.h [hunk 4]
```diff
@@ -38,8 +38,9 @@ class RSkComponentRNSVGLinearGradient final : public RSkSVGComponentNode {
   SkSVGLength x2_ = SkSVGLength(0);
   SkSVGLength y2_ = SkSVGLength(0);
   SkMatrix gradientTransforMatrix_;
-  std::vector<SkScalar> stopOffset_;
-  std::vector<SkColor> stopColor_;
+  std::vector<SkScalar> stopOffsets_;
+  std::vector<SkColor> stopColors_;
+  int gradientUnit_{0};
```
Comment (munezbn) [inline skia/RSkComponentRNSVGLinearGradient.h:48]: Should we use enum ?

File: skia/RNSVGSvgViewModule.cpp [hunk 5]
```diff
@@ -19,6 +19,7 @@ std::string RNSVGSvgViewModule::getName() {
 }
 
 auto RNSVGSvgViewModule::getConstants() -> std::map<std::string, folly::dynamic> {
+  RNS_LOG_NOT_IMPL;
```
Comment (munezbn) [inline skia/RNSVGSvgViewModule.cpp]: I dont think there is any constants required here right ? So it should just return empty object. No need of not implemented warning.

**Review by munezbn — APPROVED** [review]
Just remove RNS_NOT_IMPL from get constants

**Review by chatrapathi06 — APPROVED** [review]

**Pull request merged**